### PR TITLE
Prevent throwing 500 while finalizing an order when at least one item has become tracked

### LINKED ISSUE #10313: Making a ProductVariant tracked after it has been purchased as untracked
**Sylius version affected**: all >1.0
Based on #9435 report

Right now if we make a variant, that is used in unfulfilled orders (unpaid), tracked and then we try to either fulfil or cancel these orders we would get a 500 error. This is because when we are setting current stock to 5, it makes the variants `onHand = 5` and `onHold = 0`; then we are trying to change the order state we are trying to change the value of `onHold` (while cancelling subtract from onHold and add to onHand; and while fulfilling subtract from both onHold and onHand. Read more on inventory operations [here](https://docs.sylius.com/en/1.4/book/products/inventory.html)).

**What should happen in my opinion:**

```
As an Admin

Background:

Given there is an untracked product variant „Shoe”
And 1 item of this variant was bought in the Order #0001
And the order #0001 is in state „New"

Scenario:
When I set the "Shoe" variant as tracked with the current stock set to 5
And I save it
Then I should be informed that the on hand and on hold inventory will be modified by 1 because the Variant was bought 1 time and remains unfulfilled
And this variant’s on hand value should be 6
And this variant’s on hold value should be 1
```

### PR DESCRIPTION
**Pull request by jakubtobiasz**
| Q               | A                                                            |
|-----------------|--------------------------------------------------------------|
| Branch?         | 1.12
| Bug fix?        | yes
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | closes #10313
| License         | MIT

### CODE REVIEW

File: src/Sylius/Behat/Context/Ui/Admin/ManagingOrdersContext.php [hunk 1]
```diff
@@ -531,6 +531,17 @@ public function iShouldBeNotifiedThatTheOrderSPaymentHasBeenSuccessfullyComplete
         );
     }
 
+    /**
+     * @Then I should be notified that the order's payment could not be finalized due to insufficient stock
+     */
+    public function iShouldBeNotifiedThatTheOrderSPaymentCouldNotBeFinalizedDueToInsufficientStock()
```
Comment (GSadee) [inline src/Sylius/Behat/Context/Ui/Admin/ManagingOrdersContext.php:537]:
```suggestion
    public function iShouldBeNotifiedThatTheOrderSPaymentCouldNotBeFinalizedDueToInsufficientStock(): void
```

**Review by GSadee — APPROVED** [review]

**Review by SzymonFilipek — APPROVED** [review]
Changes passed QA check

**Comment by GSadee** [comment]
Thank you, Jacob! :1st_place_medal:

**Pull request merged**